.github/workflows/maven-build.yml

```diff
@@ -69,3 +75,18 @@ jobs:
       
     - name: Build with Maven
       run: mvn ${{ env.debugFlag }} --batch-mode  --file pom.xml clean ${{ inputs.mavenPhases }}
+
+    - name: Upload Checkstyle reports to GitHub
+      uses: jwgmeligmeyling/checkstyle-github-action@v1.2
```

Comment:
The checkstyle action fails the build when there are checkstyle errors. I prefer not to fail the build in GitHub Actions, but rather let the developers fail the build in Maven itself (i.e. when the project is fully compliant with the defined rules, then we run `checkstyle:check` in the build to fail the build if an error is found).

Or it could be another option?

![image](https://github.com/sentrysoftware/workflows/assets/32521698/587bb0f4-3646-4ce7-b47c-1e7790327b6b)

Reply:
I guess this is more related to GitHub settings, the build didn't failed.
In my test repo I also have checkstyle issues but the PR is not marked has blocked

![image](https://github.com/sentrysoftware/workflows/assets/48674878/e8aa1d36-b293-4197-a560-86abbe12249d)

Reply:
Yes, but we definitely want failed build checks to fail the build and prevent the merge of the PR. I don't understand why the checkstyle check fails, while PMD and spotbugs don't fail, even though they are reporting problems just like checkstyle.